src/rgw/rgw_rados.h

```diff
@@ -254,6 +254,14 @@ class RGWObjManifest {
                              as object might have been copied across buckets */
   map<uint64_t, RGWObjManifestRule> rules;
 
+  /* If this object is copied from another versioned object, @src_instance 
+   * is used to store src's version_id. 
+   * 
+   * @copied_obj is used to indicate whether @src_instance is valid.
+   * */
+  string src_instance; 
+  bool copied_obj;
```

Comment:
@yehudasa 
`The following example shows that we can not decide dest object's src_instance value 
only by source object's src_instance. All the buckets in this example share the same
data pool.

+------------------+            +--------------------+           +---------------------------+ 
|Bucket A          |            |   Bucket B         |           |   Bucket C                |   
|                  |            |                    |           |                           |  
|versioning CLOSED |            | versioning ENABLED |           | versioning ENABLED        |   
|------------------+            |--------------------+           |---------------------------+      
|Object AA         | - copy1 -> |Object BB           | - copy2 ->|Object CC                  |         
|                  |            |                    |           |                           | 
|instance = ""     |            |instance = "abcd123"|           |instance = "xyz789"        |    
|src_instance = "" |            |src_instance = ""   |           |src_instance = "???abcd123"|   
+------------------+            +--------------------+           +---------------------------+   

copy1:

Because object AA's src_instance is empty, use ""(object AA's instance) as object BB's src_instance.

It's OK.

copy2:

Because object BB's src_instance is empty, use "abcd123"(object BB's instance) as object CC's src_instance.

This will cause an ERROR. Because we should use "" as object CC's src_instance, not "abcd123".

-------------

So with 'coped_obj' we can decide if src object is a object generated by a copy operation. 
If source object is not a copied object, we must copy its 'instance' to dest object's 'src_instance'. 
If source object is a copied object, we can directly copy its 'src_instance' to dest object' 'src_instance'.

In the copy chain, only the first time copy need to copy source object's 'instance' to dest object's 
'src_instance'.

AA --copy*--> BB --copy--> CC --copy--> DD ...     `